proxy/src/main/java/net/md_5/bungee/BungeeBossBar.java

```diff
@@ -0,0 +1,285 @@
+package net.md_5.bungee;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.UUID;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import net.md_5.bungee.api.boss.BossBarColor;
+import net.md_5.bungee.api.boss.BossBarDivision;
+import net.md_5.bungee.api.boss.BossBarFlag;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+import net.md_5.bungee.chat.ComponentSerializer;
+import net.md_5.bungee.protocol.DefinedPacket;
+import net.md_5.bungee.protocol.ProtocolConstants;
+import net.md_5.bungee.protocol.packet.BossBar;
+
+@ToString
+@EqualsAndHashCode
+public class BungeeBossBar implements net.md_5.bungee.api.boss.BossBar
+{
+
+    @Getter
+    private BaseComponent[] title;
+    @Getter
+    private BossBarColor color;
+    @Getter
+    private BossBarDivision division;
+    @Getter
+    private float health;
+    @Getter
+    private boolean visible;
+
+    private List<ProxiedPlayer> players;
+    private EnumSet<BossBarFlag> flags;
+
+    @ToString.Exclude
+    private UUID uuid;
+
+    public BungeeBossBar(BaseComponent[] title, BossBarColor color, BossBarDivision division, float health)
+    {
+        this.title = Preconditions.checkNotNull( title, "title" );
+        this.color = Preconditions.checkNotNull( color, "color" );
+        this.division = Preconditions.checkNotNull( division, "division" );
+        Preconditions.checkArgument( 0 <= health && health <= 1, "Health may not be lower than 0 or greater than 1" );
+        this.health = health;
+        this.visible = true;
+        this.players = new ArrayList<>();
+        this.flags = EnumSet.noneOf( BossBarFlag.class );
+        this.uuid = UUID.randomUUID();
+    }
+
+    @Override
+    public boolean addPlayer(ProxiedPlayer player)
+    {
+        Preconditions.checkNotNull( player, "player" );
+        if ( !players.contains( player ) )
+        {
+            players.add( player );
```

Comment:
Return false

Reply:
It is already there??? Wtf dude

Reply:
The return value should be the packet sent success, not the contains. It looks like you don't read the docs, do you?

Reply:
Yes but why send the packet again if a plugin tries adding the same player twice?